### ISSUE TITLE
Can't create sequence

### ISSUE THREAD

**Issue by pcrocker-pivotal** [mikefarah/yaml#8]
I'm trying to create this:
```
applications:
- name: wasabi
- name: oscar
```

By doing this:
```
touch manifest.yml
yaml w -i manifest.yml applications[0].name wasabi
yaml w -i manifest.yml applications[1].name oscar
```

but... the third line (`yaml w -i manifest.yml applications[1].name oscar`) results in this:
```
# yaml w -i manifest.yml applications[1].name oscar
panic: runtime error: index out of range

goroutine 1 [running]:
panic(0x5e3500, 0xc420010080)
	/usr/local/Cellar/go/1.7.5/libexec/src/runtime/panic.go:500 +0x1a1
main.writeArray(0x5cb680, 0xc42000ec60, 0xc4200487d0, 0x2, 0x3, 0x5cfbc0, 0xc420010fd0, 0xc4200b1980, 0x4d3040, 0xc420066c90)
	/Users/mikefarah/dev/go/src/github.com/mikefarah/yaml/data_navigator.go:85 +0x5c3
main.updatedChildValue(0x5cb680, 0xc42000ec60, 0xc4200487d0, 0x2, 0x3, 0x5cfbc0, 0xc420010fd0, 0x4146be, 0xc42000ece0)
	/Users/mikefarah/dev/go/src/github.com/mikefarah/yaml/data_navigator.go:61 +0x1a6
main.writeMap(0x5da5a0, 0xc42000ece0, 0xc4200487c0, 0x3, 0x4, 0x5cfbc0, 0xc420010fd0, 0xc4200b1b47, 0xc4200b1b60, 0x52f630)
	/Users/mikefarah/dev/go/src/github.com/mikefarah/yaml/data_navigator.go:44 +0x38c
main.updateYaml(0xc420048740, 0x3, 0x4, 0xc420010fb5, 0xc4200b1c77)
	/Users/mikefarah/dev/go/src/github.com/mikefarah/yaml/yaml.go:191 +0x2de
main.writeProperty(0xc4201021e0, 0xc420048740, 0x3, 0x4)
	/Users/mikefarah/dev/go/src/github.com/mikefarah/yaml/yaml.go:167 +0x58
github.com/mikefarah/yaml/Godeps/_workspace/src/github.com/spf13/cobra.(*Command).execute(0xc4201021e0, 0xc4200485c0, 0x4, 0x4, 0xc4201021e0, 0xc4200485c0)
	/Users/mikefarah/dev/go/src/github.com/mikefarah/yaml/Godeps/_workspace/src/github.com/spf13/cobra/command.go:547 +0x411
github.com/mikefarah/yaml/Godeps/_workspace/src/github.com/spf13/cobra.(*Command).Execute(0xc4201025a0, 0xc4200b1f10, 0x3)
	/Users/mikefarah/dev/go/src/github.com/mikefarah/yaml/Godeps/_workspace/src/github.com/spf13/cobra/command.go:630 +0x398
main.main()
	/Users/mikefarah/dev/go/src/github.com/mikefarah/yaml/yaml.go:41 +0x449
```

I'm new to this project, so not sure if i'm doing something wrong... or if I've hit a bug.

**Comment by mikefarah** [comment]
Looks like a bug, I'll look into it :)

**Comment by mikefarah** [comment]
Fixed in 1.9!

Also, here's some alternative syntax:

``
yaml n 'applications[0].name' wasabi | yaml w - 'applications[1].name' oscar > manifest.yml
``

Still a bit clumsy, would be nice to make it easier to construct lists but anyway.

**Comment by pcrocker-pivotal** [comment]
Thanks!!
